runtime/src/lib.rs

```diff
@@ -238,11 +358,460 @@ parameter_types! {
+
+					trait_struct.records.windows(2).all(|window| {
+						let (record_1, record_2) = (&window[0], &window[1]);
+						let (Ok(a1), Ok(a2)) = (get_utf8_string(&record_1.0), get_utf8_string(&record_2.0)) else { // `a1` is short for `attribute_1`, `a2` is short for `attribute_2`
+							return false;
+						};
+
+						let (Some(first_char_a1), Some(first_char_a2)) = (a1.chars().next(), a2.chars().next()) else {
+							return false;
+						};
+
+						(first_char_a1.is_alphabetic() && first_char_a2.is_alphabetic()) // ensure first character is an alphabet
+							&&
+							(a1 == a1.to_lowercase() && a2 == a2.to_lowercase()) // ensure lowercase
+							&&
+							a1 <= a2 // ensure sorted. Note: "Strings are ordered lexicographically by their byte values ... This is not necessarily the same as “alphabetical” order, which varies by language and locale". Source: https://doc.rust-lang.org/std/primitive.str.html#impl-Ord-for-str
+							&&
+							a1 != a2 // ensure no duplicates
+					})
+				},
+			},
+			Categories::Shards(shards_script_info_struct) => {
+				let format = shards_script_info_struct.format;
+				let requiring = &shards_script_info_struct.requiring;
+				let implementing = &shards_script_info_struct.implementing;
+
+				let all_required_trait_impls_found = requiring.iter().all(|shards_trait| {
+					proto_references.iter().any(|proto| {
+						if let Some(trait_impls) = pallet_protos::TraitImplsByShard::<Runtime>::get(proto) {
+							trait_impls.contains(shards_trait)
+						} else {
+							false
+						}
+					})
+				});
+
+				let all_traits_implemented_in_this_shards = implementing.iter().all(|_shards_trait| {
+					match format {
+						ShardsFormat::Edn => {
+							true
+						},
+						ShardsFormat::Binary => {
+							true
+						},
+					}
+				});
+
+				all_required_trait_impls_found && all_traits_implemented_in_this_shards
+			},
+			Categories::Audio(sub_categories) => match sub_categories {
+				AudioCategories::OggFile => infer::is(data, "ogg"),  // TODO Review - We are not checking for other OGG file extensions https://en.wikipedia.org/wiki/Ogg
+				AudioCategories::Mp3File => infer::is(data, "mp3"),
+			},
+			Categories::Texture(sub_categories) => match sub_categories {
+				TextureCategories::PngFile => infer::is(data, "png"), // png_decoder::decode(&data[..]).is_ok(),
+				TextureCategories::JpgFile => infer::is(data, "jpg"), // TODO Review - we do not include ".jpeg" images, only ".jpg" images
+			},
+			Categories::Vector(sub_categories) => match sub_categories {
+				VectorCategories::SvgFile => false,
+				VectorCategories::TtfFile => infer::is(data, "ttf"), // ttf_parser::Face::parse(&data[..], 0).is_ok(),
+			},
+			Categories::Video(sub_categories) => match sub_categories {
+				VideoCategories::MkvFile => infer::is(data, "mkv"),
+				VideoCategories::Mp4File => infer::is(data, "mp4"),
+			},
+			Categories::Model(sub_categories) => match sub_categories {
+				ModelCategories::GltfFile => true,
```

Comment:
this makes this whole PR worthless, let's do false 👍

Reply:
for everything I just made `true`, right?

Reply:
what do you mean?
I mean a malicious party can select gltf and upload anything.

Reply:
okay nevermind. I thought you wanted me to revert all the comments earlier where you said "put true". So I'll just make this `false`

Reply:
done